lib/components/home/artist_rails/artist_card.js

```diff
@@ -49,7 +56,7 @@ class ArtistCard extends React.Component {
   }
 
   setFollowStatus = (status: boolean) => {
-    return new Promise((followStatusChanged) => {
+    return new Promise((followStatusChanged: boolean) => {
```

Comment:
OK, so, to set the state with it's known types, we need to ensure that `followStatusChanged` is a boolean, to do that we have to make sure the function is typed correctly. To do that we have to make the Promise object be a `Promise<boolean>` by doing the `import { Promise } from 'react-native'` above we let flow know that a Promise can be bound to a specific class, which Flow then infers to be a bool based on the state's definition above.

```
    return new Promise((followStatusChanged: boolean) => {
        this.setState({
          following: status,
          processingChange: false,
         followStatusChanged
       })
```